bodhi/server/consumers/masher.py

```diff
@@ -721,6 +735,37 @@ def sanity_check_repo(self):
 
         return True
 
+    def wait_for_repo_signature(self):
+        """Wait for a repo signature to appear if configured as such."""
+        if not config.get('wait_for_repo_sig'):
+            self.log.info('Not waiting for a repo signature')
+            return
+        sigpaths = {}
+        for arch in os.listdir(os.path.join(self.path, self.id)):
+            sigpaths.append(os.path.join(self.path, self.id, arch, 'repodata',
+                                         'repomd.xml.asc'))
+        self.log.info('Waiting for signatures in %s' % sigpaths)
+        notifications.publish(
+            topic="reposig.wait",
+            msg=dict(repo=self.id, agent=self.agent),
+            force=True,
+        )
+        while True:
+            got_all = True
+            for path in sigpaths:
+                if not os.path.exists(path):
+                    got_all = False
+            if got_all:
+                self.log.info('All signatures were created')
+            else:
+                self.log.info('Waiting')
+                time.sleep(5)
```

Comment:
Should we have some huge amount of time that we give up and error out after? Like, if it hasn't happened in 10 hours or something arbitrary?

Reply:
Well, if we at some point enable this on clients, and we send any repo without signature, that repo will be useless.
So when we turn this on, we should make it required to proceed.

Reply:
Also, during the initial time while it's not yet enabled for clients, if for whatever reason the signature doesn't get generated, we can just create the files by hand.